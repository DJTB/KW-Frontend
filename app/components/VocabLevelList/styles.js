import styled from 'styled-components';

import { resetList } from 'shared/styles/utils';
import { containerGutter } from 'shared/styles/layout';

export const Ul = styled.ul`
  ${resetList}
  ${containerGutter}
`;
