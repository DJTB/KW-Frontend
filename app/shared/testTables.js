import { random } from 'lodash';

/**
 * These all assume that the data has been serialized to preferred format in the front end already!
 * This is not the API response shape.
 */

export const levels = [
  {
    level: 1,
    unlocked: false,
    count: 43,
  },
  {
    level: 2,
    unlocked: true,
    count: 90,
  },
  {
    level: 3,
    unlocked: true,
    count: 69,
  },
  {
    level: 4,
    unlocked: false,
    count: 105,
  },
  {
    level: 5,
    unlocked: true,
    count: 126,
  },
  {
    level: 6,
    unlocked: true,
    count: 118,
  },
  {
    level: 7,
    unlocked: true,
    count: 101,
  },
  {
    level: 8,
    unlocked: false,
    count: 136,
  },
  {
    level: 9,
    unlocked: false,
    count: 119,
  },
  {
    level: 10,
    unlocked: false,
    count: 118,
  },
  {
    level: 11,
    unlocked: false,
    count: 126,
  },
  {
    level: 12,
    unlocked: false,
    count: 126,
  },
  {
    level: 13,
    unlocked: false,
    count: 115,
  },
  {
    level: 14,
    unlocked: false,
    count: 116,
  },
  {
    level: 15,
    unlocked: false,
    count: 100,
  },
  {
    level: 16,
    unlocked: false,
    count: 117,
  },
  {
    level: 17,
    unlocked: false,
    count: 124,
  },
  {
    level: 18,
    unlocked: false,
    count: 135,
  },
  {
    level: 19,
    unlocked: false,
    count: 106,
  },
  {
    level: 20,
    unlocked: false,
    count: 114,
  },
  {
    level: 21,
    unlocked: false,
    count: 106,
  },
  {
    level: 22,
    unlocked: false,
    count: 115,
  },
  {
    level: 23,
    unlocked: false,
    count: 97,
  },
  {
    level: 24,
    unlocked: false,
    count: 120,
  },
  {
    level: 25,
    unlocked: false,
    count: 100,
  },
  {
    level: 26,
    unlocked: false,
    count: 124,
  },
  {
    level: 27,
    unlocked: false,
    count: 106,
  },
  {
    level: 28,
    unlocked: false,
    count: 116,
  },
  {
    level: 29,
    unlocked: false,
    count: 107,
  },
  {
    level: 30,
    unlocked: false,
    count: 103,
  },
  {
    level: 31,
    unlocked: false,
    count: 113,
  },
  {
    level: 32,
    unlocked: false,
    count: 111,
  },
  {
    level: 33,
    unlocked: false,
    count: 103,
  },
  {
    level: 34,
    unlocked: false,
    count: 124,
  },
  {
    level: 35,
    unlocked: false,
    count: 100,
  },
  {
    level: 36,
    unlocked: false,
    count: 98,
  },
  {
    level: 37,
    unlocked: false,
    count: 112,
  },
  {
    level: 38,
    unlocked: false,
    count: 100,
  },
  {
    level: 39,
    unlocked: false,
    count: 101,
  },
  {
    level: 40,
    unlocked: false,
    count: 116,
  },
  {
    level: 41,
    unlocked: false,
    count: 107,
  },
  {
    level: 42,
    unlocked: false,
    count: 96,
  },
  {
    level: 43,
    unlocked: false,
    count: 108,
  },
  {
    level: 44,
    unlocked: false,
    count: 103,
  },
  {
    level: 45,
    unlocked: false,
    count: 96,
  },
  {
    level: 46,
    unlocked: false,
    count: 103,
  },
  {
    level: 47,
    unlocked: false,
    count: 94,
  },
  {
    level: 48,
    unlocked: false,
    count: 104,
  },
  {
    level: 49,
    unlocked: false,
    count: 104,
  },
  {
    level: 50,
    unlocked: false,
    count: 87,
  },
  {
    level: 51,
    unlocked: false,
    count: 75,
  },
  {
    level: 52,
    unlocked: false,
    count: 99,
  },
  {
    level: 53,
    unlocked: false,
    count: 103,
  },
  {
    level: 54,
    unlocked: false,
    count: 112,
  },
  {
    level: 55,
    unlocked: false,
    count: 76,
  },
  {
    level: 56,
    unlocked: false,
    count: 88,
  },
  {
    level: 57,
    unlocked: false,
    count: 93,
  },
  {
    level: 58,
    unlocked: false,
    count: 81,
  },
  {
    level: 59,
    unlocked: false,
    count: 79,
  },
  {
    level: 60,
    unlocked: false,
    count: 65,
  },
];

export const vocabs = [
  {
    id: 553,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['before long', 'soon'],
    readings: [
      {
        id: 582,
        character: '近々',
        kana: ['ちかぢか'],
        level: 7,
        tags: ['Adverbial Noun'],
        sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
        sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
        jlpt: 'JLPT N2',
        common: true,
      },
      {
        id: 583,
        character: '近々',
        kana: ['きんきん'],
        level: 7,
        tags: ['Adverbial Noun'],
        sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
        sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
        jlpt: 'JLPT N2',
        common: true,
      },
      {
        id: 584,
        character: '近々',
        kana: ['ちかじか'],
        level: 7,
        tags: ['Adverbial Noun'],
        sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
        sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
        jlpt: 'JLPT N2',
        common: true,
      },
      {
        id: 783,
        character: '軈て',
        kana: ['やがて'],
        level: 61,
        tags: ['Adverb'],
        sentenceEn: 'By and by he forgot that affair',
        sentenceJa: 'やがて彼はその事件のことを忘れた',
        jlpt: 'JLPT N2',
        common: true,
      },
    ],
    synonyms: [
      {
        id: 5582,
        character: '間もなく',
        kana: ['まもなく'],
        level: 22,
        tags: ['Adverb'],
        sentenceEn: 'Coming up, Jiyuugaoka Station.',
        sentenceJa: '間もなく、自由が丘駅',
        jlpt: 'JLPT N2',
        common: true,
      },
    ],
  },
  {
    id: 176,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['young lady', 'little girl', 'maiden', 'daughter', 'girl'],
    readings: [
      {
        id: 188,
        character: '少女',
        kana: ['しょうじょ'],
        level: 3,
        tags: ['Noun'],
        sentenceEn: "I am touched by the girl's acute sensitivity.",
        sentenceJa: '少女の鋭い感性に触れている',
        jlpt: 'JLPT N3',
        common: true,
      },
      {
        id: 189,
        character: '少女',
        kana: ['おとめ'],
        level: 3,
        tags: ['Noun'],
        sentenceEn: "I am touched by the girl's acute sensitivity.",
        sentenceJa: '少女の鋭い感性に触れている',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
  {
    id: 280,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['young boy', 'juvenile', 'youth', 'boy'],
    readings: [
      {
        id: 298,
        character: '少年',
        kana: ['しょうねん'],
        level: 4,
        tags: ['Noun'],
        sentenceEn: 'The shy boy murmured his name.',
        sentenceJa: 'その内気な少年はぼそぼそと自分の名前をつぶやいた全ての少年少女はを知っています',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
  {
    id: 424,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: [
      'whole life',
      'entire life',
      'a lifetime',
      'all through life',
      'one existence',
      'a generation',
      'an age',
    ],
    readings: [
      {
        id: 449,
        character: '一生',
        kana: ['いっしょう'],
        level: 5,
        tags: ['Noun', 'No adjective'],
        sentenceEn: 'She worked hard.',
        sentenceJa: '彼女は一生懸命働いた',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
  {
    id: 502,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['honest'],
    readings: [
      {
        id: 529,
        character: '正直',
        kana: ['しょうじき'],
        level: 6,
        tags: ['Noun', 'Na adjective'],
        sentenceEn: 'Him, honest? What a joke!',
        sentenceJa: '正直が聞いて呆れるよ',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
];

export const readings = {
  single: [
    {
      id: 351,
      character: '売り上げ',
      kana: ['うりあげ'],
      level: 5,
      tags: ['Noun'],
      sentenceEn: 'Sales are down this month.',
      sentenceJa: '今月は売上が落ちた',
      jlpt: 'JLPT N2',
      common: true,
    },
  ],
  sameCharacters: [
    {
      id: 582,
      character: '近々',
      kana: ['ちかぢか'],
      level: 7,
      tags: ['Noun'],
      sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
      sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
      jlpt: 'JLPT N2',
      common: true,
    },
    {
      id: 583,
      character: '近々',
      kana: ['きんきん'],
      level: 7,
      tags: ['Noun'],
      furigana: '0:きんきん',
      sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
      sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
      jlpt: 'JLPT N2',
      common: true,
    },
    {
      id: 584,
      character: '近々',
      kana: ['ちかじか'],
      level: 7,
      tags: ['Noun'],
      sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
      sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
      jlpt: 'JLPT N2',
      common: true,
    },
  ],
  suffix: [
    {
      id: 2476,
      character: '各〜',
      kana: ['かく'],
      level: 22,
      tags: ['Prefix'],
      sentenceEn: 'First, choose a representative for each team.',
      sentenceJa: 'まず始めに、各チームの代表者を決めて下さい',
      jlpt: null,
      common: null,
    },
  ],
  prefix: [
    {
      id: 6337,
      character: '〜漬け',
      kana: ['づけ'],
      level: 59,
      tags: ['Suffix'],
      sentenceEn: 'These are lightly pickled cucumbers.',
      sentenceJa: '胡瓜の浅漬けならありますよ',
      jlpt: null,
      common: null,
    },
    {
      id: 130,
      character: '〜才',
      kana: ['さい'],
      level: 2,
      tags: ['Suffix'],
      sentenceEn: 'I am five years old.',
      sentenceJa: 'わたしは５才です',
      jlpt: null,
      common: null,
    },
  ],
  mixed: [
    {
      id: 933,
      character: 'ハート形',
      kana: ['ハートがた'],
      level: 9,
      tags: ['Noun'],
      sentenceEn: 'When flamingos kiss, their heads and necks form a heart shape.',
      sentenceJa: 'フラミンゴがキスをする時、二羽の頭と首がハート形を描く',
      jlpt: null,
      common: false,
    },
    {
      id: 30,
      character: 'フランス人',
      kana: ['ふらんすじん'],
      level: 1,
      tags: ['Noun'],
      sentenceEn: 'He is French.',
      sentenceJa: '彼はフランス人です',
      jlpt: null,
      common: false,
    },
    {
      id: 3190,
      character: '迷惑メール',
      kana: ['めいわくメール'],
      level: 29,
      tags: ['Noun'],
      sentenceEn:
        'He sent about 2.2 billion spam emails over a period of around one and a half years.',
      sentenceJa: '約１年半で約億通の迷惑メールを送りました',
      jlpt: null,
      common: false,
    },
    {
      id: 6193,
      character: '烏龍茶',
      kana: ['ウーロンちゃ'],
      level: 57,
      tags: ['Noun'],
      sentenceEn: 'Oolong tea without ice, please.',
      sentenceJa: '烏龍茶を氷抜きでください',
      jlpt: null,
      common: null,
    },
    {
      id: 4875,
      character: '缶ビール',
      kana: ['かんビール'],
      level: 44,
      tags: ['Noun'],
      sentenceEn: 'Oh, this canned beer is way too good for me.',
      sentenceJa: 'ああ、この缶ビールはマジでうますぎる',
      jlpt: null,
      common: false,
    },
  ],
  onlyKana: [
    {
      id: 9000,
      common: true,
      jlpt: 'JLPT N4',
      character: 'そろそろ',
      kana: ['そろそろ'],
      tags: ['Adverb', "Adverb taking the 'to' particle"],
      sentenceJa: '「あの、かりんさん...、そろそろお開きにしませんか？」「駄目、勝ち逃げ禁止！今度こそボクが勝つの！」',
      sentenceEn:
        "“Er, Karin ..., shouldn't we call it a night soon?” “No! No quitting while ahead! Next time I'll win for sure!”",
    },
    {
      id: 9001,
      common: false,
      jlpt: null,
      character: 'マンコ',
      kana: ['マンコ'],
      tags: ['Noun'],
      sentenceJa: 'マンコいいな〜',
      sentenceEn: 'What a great pussy.',
    },
  ],
  partialKanaSentence: [
    {
      id: 8800,
      common: true,
      jlpt: 'JLPT N4',
      character: '付く',
      kana: ['つく'],
      tags: ['Transitive Verb'],
      sentenceJa: 'この「オンボード」というのは平たく言えば「グラフィックカードがついていない」ものだ',
      sentenceEn:
        'Frankly speaking, this “on board” means “doesn’t have a graphics card”.',
    },
  ],
};
