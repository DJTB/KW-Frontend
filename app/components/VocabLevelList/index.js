import React from 'react';
import PropTypes from 'prop-types';
import cuid from 'cuid';

import VocabLevel from 'components/VocabLevel';
import { Ul } from './styles';

VocabLevelList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape(VocabLevel.propTypes)),
};

VocabLevelList.defaultProps = {
  items: [],
};

function VocabLevelList({ items }) {
  // TODO: React.Children.map(CloneElement => add relevant onLock handler) ?
  return (
    <Ul>{items.map((item) => <VocabLevel key={cuid()} {...item} />)}</Ul>
  );
}

export default VocabLevelList;
