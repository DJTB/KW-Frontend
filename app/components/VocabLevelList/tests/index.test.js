import React from 'react';
import { shallow } from 'enzyme';

import { levels } from 'utils/tests/testTables';
import VocabLevelList from '../index';

describe('<VocabLevelList />', () => {
  it('should match baseline snapshot', () => {
    const renderedComponent = shallow(
      <VocabLevelList
        levels={levels.slice(30, 38)}
        userWKLevel={34}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
