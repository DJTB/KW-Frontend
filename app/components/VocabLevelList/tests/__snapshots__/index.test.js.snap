// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VocabLevelList /> should match baseline snapshot 1`] = `
<styled.ul>
  <VocabLevel
    count={113}
    handleLevelLock={[Function]}
    isActionable={true}
    isLocked={true}
    level={31}
  />
  <VocabLevel
    count={111}
    handleLevelLock={[Function]}
    isActionable={true}
    isLocked={true}
    level={32}
  />
  <VocabLevel
    count={103}
    handleLevelLock={[Function]}
    isActionable={true}
    isLocked={true}
    level={33}
  />
  <VocabLevel
    count={124}
    handleLevelLock={[Function]}
    isActionable={true}
    isLocked={true}
    level={34}
  />
  <VocabLevel
    count={100}
    handleLevelLock={[Function]}
    isActionable={false}
    isLocked={true}
    level={35}
  />
  <VocabLevel
    count={98}
    handleLevelLock={[Function]}
    isActionable={false}
    isLocked={true}
    level={36}
  />
  <VocabLevel
    count={112}
    handleLevelLock={[Function]}
    isActionable={false}
    isLocked={true}
    level={37}
  />
  <VocabLevel
    count={100}
    handleLevelLock={[Function]}
    isActionable={false}
    isLocked={true}
    level={38}
  />
</styled.ul>
`;
